woocommerce/woocommerce-android
# [REST API] Log user out if user is unauthorized to generate application password

Closes: #8221


### Description
This PR simply monitors the application password generation failures, and log the user out when a failure due to a `401` error happens.

### Testing instructions
1. Make sure you are assigned the Treatment variant of the REST API experiment (either by [hardcoding](https://github.com/woocommerce/woocommerce-android/blob/9841083ed877c1846ca66d43a44841ecfb320837/WooCommerce/src/main/kotlin/com/woocommerce/android/experiment/RESTAPILoginExperiment.kt#L43) it or by setting a test device)
2. Open the app, then sign in using site credentials.
3. Open your site, then change the user password, and revoke the application password.
4. Go back to the app, then do tasks to trigger API requests.
5. Confirm the app logs you out.
